src/components/TheHeader.vue

```diff
@@ -0,0 +1,81 @@
+<script lang="ts" setup>
+import MoonIcon from "@/assets/icons/MoonIcon.vue";
+import { ECountriesMode } from "@/types/countries";
+import type { Ref } from "vue";
+import { computed, ref } from "vue";
+
+const themeMode: Ref<ECountriesMode> = ref(ECountriesMode.LIGHT_MODE);
+const themeModeTile = computed(() =>
+  themeMode.value === ECountriesMode.LIGHT_MODE ? "Light Mode" : "Dark Mode"
+);
+
+const changeTheme = () => {
+  themeMode.value === ECountriesMode.LIGHT_MODE
+    ? (themeMode.value = ECountriesMode.DARK_MODE)
+    : (themeMode.value = ECountriesMode.LIGHT_MODE);
+
+  document.querySelector(":root")?.classList.toggle("dark-theme");
+};
+</script>
+
+<template>
+  <header class="c-header">
+    <div class="c-header__elements-wrapper">
+      <RouterLink class="c-header__title" to="/"
+        >Where is the world?
+      </RouterLink>
+      <button class="c-header__theme-container" @click="changeTheme()">
+        <MoonIcon class="c-header__moon-icon" />
+        <span class="c-header__theme-mode-title">{{ themeModeTile }}</span>
+      </button>
+    </div>
+  </header>
+</template>
+
+<style lang="scss" scoped>
+.c-header {
+  height: 8.2rem;
+  background-color: var(--header-color);
+  box-shadow: var(--shadow);
+  display: flex;
+  align-items: center;
+
+  &__elements-wrapper {
+    margin: 0 auto;
+    padding: 0 3.2rem;
+    width: 124rem;
```

Comment:
ustawianie width na sztywno to nie jest dobry pomysl, zastąp sobie max-width

Reply:
jeżeli mam użyć max-width, to muszę nadać dodatkowo width na 100%, inaczej nie będzie mi to funkcjonowało. W przypadku samego max-width nie dziala justify-content

Reply:
ale czemu nie będzie Ci to funkcjonowało? Wartość domyślna dla width to auto, czyli z reguły 100% rodzica (w dużym uproszczeniu)

Reply:
https://codepen.io/web-lizzard/pen/YzjNWZy